lib/THC/THCStream.c

```diff
@@ -13,18 +16,38 @@ THCStream* THCStream_new(int flags)
   return self;
 }
 
+THC_API THCStream* THCStream_defaultStream(int device)
+{
+  // default streams aren't refcounted
+  THAssert(device >= 0 && device < MAX_DEVICES);
+  THCStream* stream = &default_streams[device];
+  stream->device = device;
```

Comment:
I don't know if this would ever practically go wrong, but my reading of the C standard is it's not technically safe.